app/models/digital_object/base.rb

```diff
@@ -28,7 +28,8 @@ class Base
     metadata_attribute :serialization_version, Hyacinth::DigitalObject::TypeDef::String.new.default(-> { SERIALIZATION_VERSION })
     metadata_attribute :uid, Hyacinth::DigitalObject::TypeDef::String.new
     metadata_attribute :doi, Hyacinth::DigitalObject::TypeDef::String.new
-    metadata_attribute :digital_object_type, Hyacinth::DigitalObject::TypeDef::String.new
+    # constrain type to the keys for registered type classes
+    metadata_attribute :digital_object_type, Hyacinth::DigitalObject::TypeDef::String.new.constrained_to(Hyacinth.config.digital_object_types)
```

Comment:
Can we make this a DigitalObject validation (in `DigitalObjectConcerns::Validations`) instead of something on the TypeDef object?  I originally had value constraints on the TypeDef and intentionally removed them because I didn't like having some validation in the TypeDef, but most other validations in `DigitalObjectConcerns::Validations`.  Note that validations are pretty much non-existent right now because adding them in `DigitalObjectConcerns::Validations` is a TODO.